runtime/recipe/view-connection.js

```diff
@@ -58,14 +60,29 @@ class ViewConnection {
   get recipe() { return this._recipe; }
   get name() { return this._name; } // Parameter name?
   get tags() { return this._tags; }
-  get type() { return this._type; }
+  get type() {
+    if (this._type)
+      return this._type;
+    return this._rawType;
+  }
+  get rawType() {
+    return this._rawType;
+  }
   get direction() { return this._direction; } // in/out
   get view() { return this._view; } // View?
   get particle() { return this._particle; } // never null
 
   set tags(tags) { this._tags = tags; }
-  set type(type) { this._type = type;}
-  set direction(direction) { this._direction = direction; }
+  set type(type) {
```

Comment:
I'm going to leave this as-is for now. What I was going for here is that everything external to recipe+typeChecker sees the type as a unitary thing. Only the typeChecker (so it can override with a resolved type) and the recipe (so it can do validity checking on the spec) know different.

However, it's not the most obvious approach in the world, so I'd like to talk it through with you f2f.